lib/core.js

```diff
@@ -1355,12 +1355,65 @@
         }
     }
 
-    function generateProviderOptionsFunc(options) {
+    function normalizeValue(value) {
+        if (value === 'true') {
+            return true;
+        }
+        if (value === 'false') {
+            return false;
+        }
+        if (value.match && value.match(/^\d+$/)) {
+            return parseInt(value);
+        }
+        if (value.match && value.match(/^(\d+)?\.\d+$/)) {
+            return parseFloat(value);
+        }
+        return value;
+    }
+
+    var _RE = /^_.+/;
+
+    function getProviderOptionsByUrl(url) {
+        /*
+        Convert '?_option=value' to 
+        {
+            options: value
+        }
+        */
+
+        try {
+            var urlSearchParams = new URL(url).searchParams;
+        } catch(ex) {
+            console.log('Error parsing url', url);
+            return;
+        }
+
+        var result = {};
+
+        urlSearchParams.forEach((value, key) => {
+            if (key.length > 1 && _RE.test(key)) {
```

Comment:
`key.length >1` is redundant here if `_RE = /^_.+/;`

Reply:
I think `key.length` is faster, so its only for performance.